### PR TITLE
Update embeddings.py

### PR DESCRIPTION
**Pull request by soma2000-lang**
## Relevant Issue
#138

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@divyanshugit  Please look into it.

File: fedrec/modules/embeddings.py [hunk 1]
```diff
@@ -149,28 +242,21 @@ class QREmbeddingBag(nn.Module):
     using the quotient of the indices and the other using the remainder
     of the indices, without instantiating the intermediate embeddings,
     then performsan operation to combine these.
-
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:152]: Why are you removing these lines?

File: fedrec/modules/embeddings.py [hunk 2]
```diff
@@ -24,13 +24,19 @@ def md_solver(n, alpha, d0=None, B=None, round_dim=True, k=None):
...
-    Inputs:
+    
+    Arguments
+    ----------
     n -- (torch.LongTensor) ; Vector of num of rows for each embedding matrix
     alpha -- (torch.FloatTensor); Scalar, non-negative, controls dim. skew
     d0 -- (torch.FloatTensor); Scalar, baseline embedding dimension
     B -- (torch.FloatTensor); Scalar, parameter budget for embedding layer
     round_dim -- (bool); flag for rounding dims to nearest pow of 2
     k -- (torch.LongTensor) ; Vector of average number of queries per inference
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:35]: It will look good.  If you can write as ``` n(torch.LongTensor) - ```  instead of  ```n -- (torch.LongTensor);```  and make the proper indentation as well.
```suggestion
        n -- (torch.LongTensor) ; Vector of num of rows for each embedding matrix
        alpha -- (torch.FloatTensor); Scalar, non-negative, controls dim. skew
        d0 -- (torch.FloatTensor); Scalar, baseline embedding dimension
        B -- (torch.FloatTensor); Scalar, parameter budget for embedding layer
        round_dim -- (bool); flag for rounding dims to nearest pow of 2
        k -- (torch.LongTensor) ; Vector of average number of queries per inference
```

File: fedrec/modules/embeddings.py [hunk 3]
```diff
@@ -99,12 +156,27 @@ def __init__(
 
 @registry.load("embedding", "pr_emb")
 class PrEmbeddingBag(nn.Module):
+    '''It is like adding a extra rapper layer upon the embedding.
+        
+        ----
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:161]: Unnecessary lines.

File: fedrec/modules/embeddings.py [hunk 4]
```diff
@@ -99,12 +156,27 @@ def __init__(
...
                  index=-1,
                  init=False):
+    '''Base class for all neural network modules.Used for the function definition
+   
+   Arguments
+   -----------------
+    num_embeddings:(int) – size of the dictionary of embedding.
+    embedding_dim:(int) – the size of each embedding vector.
+    base_dim-The base dimension.
+    index:(int) -the particular index.
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:178]: Indentation issue.

File: fedrec/modules/embeddings.py [hunk 5]
```diff
@@ -44,9 +50,24 @@ def md_solver(n, alpha, d0=None, B=None, round_dim=True, k=None):
...
+    n -- (torch.LongTensor) ; Vector of num of rows for each embedding matrix
+    alpha -- (torch.FloatTensor); Scalar, non-negative, controls dim. skew
+    d0 -- (torch.FloatTensor); Scalar, baseline embedding dimension
+    B -- (torch.FloatTensor); Scalar, parameter budget for embedding layer
+    
+    
+    Returns
+    --------
+    torch.round(d).type(torch.long))(Tensor)-Rounds elements of input to the nearest integer.
+    '''
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:67]: In this comment block, there are multiple issues:
- Indentation
- It will look good.  If you can write as ``` n(torch.LongTensor) - ```  instead of  ```n -- (torch.LongTensor);```

File: fedrec/modules/embeddings.py [hunk 6]
```diff
@@ -339,6 +431,16 @@ def forward(self, input, offsets=None, per_sample_weights=None):
...
     def extra_repr(self):
+        """ Set the extra representation of the module
+            To print customized extra information, you should 
+            re-implement this method in your own modules. Both single-line and multi-line strings are acceptable.
+            
+            
+            Returns
+            ----------------------
+            s.format()
+            """
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:442]: Please use, proper indentation.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
@soma2000-lang Can you please refrain from adding or removing unnecessary lines while writing the inline docs?

**Comment by soma2000-lang** [comment]
> @soma2000-lang Can you please refrain from adding or removing unnecessary lines while writing the inline docs?

Sorry @divyanshugit  will correct it soon.

File: fedrec/modules/embeddings.py [hunk 7]
```diff
@@ -23,14 +23,20 @@
...
     '''
     An external facing function call for mixed-dimension assignment
-    with the alpha power temperature heuristic
-    Inputs:
-    n -- (torch.LongTensor) ; Vector of num of rows for each embedding matrix
+    with the alpha power temperature heuristic.
+
+    Arguments
+    ----------
+    n(torch.LongTensor); Vector of num of rows for each embedding matrix
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:30]: Can you please write it like this:
n(torch.LongTensor) -
```suggestion
    n(torch.LongTensor) - Vector of num of rows for each embedding matrix
```
and follow the same across the file.

File: fedrec/modules/embeddings.py [hunk 8]
```diff
@@ -137,6 +215,26 @@ def __init__(self,
                     torch.tensor(W, requires_grad=True))
 
     def forward(self, input, offsets=None, per_sample_weights=None):
+        """
+        Forward pass of EmbeddingBag.
+
+        Arguments
+        --------------
+        input (Tensor)  Tensor containing bags of indices into the embedding matrix.
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:223]:
```suggestion
        input(Tensor)- Tensor containing bags of indices into the embedding matrix.
```
Please write the arguments/parameters in this way across the file.

File: fedrec/modules/embeddings.py [hunk 9]
```diff
@@ -264,6 +352,11 @@ class QREmbeddingBag(nn.Module):
                      'scale_grad_by_freq',
                      'mode', 'sparse']
 
+    def __init__(self, num_embeddings, embedding_dim, num_collisions,
+                 operation='mult', max_norm=None, norm_type=2.,
+                 scale_grad_by_freq=False, mode='mean', sparse=False,
+                 _weight=None):
+        """__init__ function is called every time an object is created from a class."""
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:359]: What are you trying to do over here?

File: fedrec/modules/embeddings.py [hunk 10]
```diff
@@ -339,12 +448,23 @@ def forward(self, input, offsets=None, per_sample_weights=None):
...
+        def extra_repr(self):
+            s = '{num_embeddings}, {embedding_dim}'
+            if self.max_norm is not None:
+                s += ', max_norm={max_norm}'
+            if self.norm_type != 2:
+                s += ', norm_type={norm_type}'
+            if self.scale_grad_by_freq is not False:
+                s += ', scale_grad_by_freq={scale_grad_by_freq}'
+            s += ', mode={mode}'
+            return s.format(**self.__dict__)
```
Comment (divyanshugit) [inline fedrec/modules/embeddings.py:470]: What is happening over here?

**Comment by divyanshugit** [comment]
You can write up to 79 characters in a single line. Please try to follow it. You can find log over [here](https://github.com/NimbleEdge/EnvisEdge/runs/6421376291?check_suite_focus=true)
